drivers/wifi/nrfwifi/src/wpa_supp_if.c

```diff
@@ -0,0 +1,2761 @@
+			NRF_WIFI_CIPHER_SUITE_VALID;
+	}
+
+	if (params->local_state_change) {
+		auth_info.nrf_wifi_flags |= NRF_WIFI_CMD_AUTHENTICATE_LOCAL_STATE_CHANGE;
+	}
+
+	status = nrf_wifi_fmac_auth(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, &auth_info);
+
+	if (status != NRF_WIFI_STATUS_SUCCESS) {
+		LOG_ERR("%s: MLME command failed (auth): count=%d ret=%d", __func__, count, ret);
+		count++;
+		ret = -1;
+	} else {
+		LOG_DBG("%s:Authentication request sent successfully", __func__);
+		ret = 0;
+	}
+out:
+	k_mutex_unlock(&vif_ctx_zep->vif_lock);
+	return ret;
+}
+
+int nrf_wifi_wpa_supp_associate(void *if_priv, struct wpa_driver_associate_params *params)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	struct nrf_wifi_vif_ctx_zep *vif_ctx_zep = NULL;
+	struct nrf_wifi_ctx_zep *rpu_ctx_zep = NULL;
+	struct nrf_wifi_umac_assoc_info assoc_info;
+	int ret = -1;
+
+	if ((!if_priv) || (!params)) {
+		LOG_ERR("%s: Invalid params", __func__);
+		return ret;
+	}
+
+	vif_ctx_zep = if_priv;
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+	if (!rpu_ctx_zep->rpu_ctx) {
+		LOG_DBG("%s: RPU context not initialized", __func__);
+		return ret;
+	}
+
+	k_mutex_lock(&vif_ctx_zep->vif_lock, K_FOREVER);
+	if (!rpu_ctx_zep->rpu_ctx) {
+		LOG_DBG("%s: RPU context not initialized", __func__);
+		goto out;
+	}
+
+	memset(&assoc_info, 0, sizeof(assoc_info));
+
+	if (params->bssid) {
+		memcpy(assoc_info.nrf_wifi_bssid, params->bssid, sizeof(assoc_info.nrf_wifi_bssid));
+	}
+
+	if (params->freq.freq) {
+		assoc_info.center_frequency = params->freq.freq;
+		vif_ctx_zep->assoc_freq = params->freq.freq;
+	} else {
+		vif_ctx_zep->assoc_freq = 0;
+	}
+
+	if (params->prev_bssid) {
+		assoc_info.prev_bssid_flag = 1;
+		memcpy(assoc_info.prev_bssid, params->prev_bssid, ETH_ALEN);
+	}
+
+	if (params->ssid) {
+		assoc_info.ssid.nrf_wifi_ssid_len = params->ssid_len;
+
+		memcpy(assoc_info.ssid.nrf_wifi_ssid, params->ssid, params->ssid_len);
+
+	}
+
+	if (params->wpa_ie) {
+		assoc_info.wpa_ie.ie_len = params->wpa_ie_len;
+		memcpy(assoc_info.wpa_ie.ie, params->wpa_ie, params->wpa_ie_len);
+	}
+
+	assoc_info.control_port = 1;
+
+	if (params->mgmt_frame_protection == MGMT_FRAME_PROTECTION_REQUIRED) {
+		assoc_info.use_mfp = NRF_WIFI_MFP_REQUIRED;
+	}
+
+	if (params->bss_max_idle_period) {
+		assoc_info.bss_max_idle_time = params->bss_max_idle_period;
+	}
+
+	status = nrf_wifi_fmac_assoc(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, &assoc_info);
+
+	if (status != NRF_WIFI_STATUS_SUCCESS) {
+		LOG_ERR("%s: MLME command failed (assoc)", __func__);
+	} else {
+		LOG_DBG("%s: Association request sent successfully", __func__);
+		ret = 0;
+	}
+
+out:
+	k_mutex_unlock(&vif_ctx_zep->vif_lock);
+	return ret;
+}
+
+int nrf_wifi_wpa_supp_set_key(void *if_priv, const unsigned char *ifname, enum wpa_alg alg,
+			      const unsigned char *addr, int key_idx, int set_tx,
+			      const unsigned char *seq, size_t seq_len, const unsigned char *key,
+			      size_t key_len)
```

Comment:
The signature of this function does not match https://github.com/zephyrproject-rtos/hostap/blob/a90df86d7c596a5367ff70c2b50c7f599e6636f3/src/drivers/driver_zephyr.h#L198-L208 resulting in the follwing warning

```
/home/nordic/Documents/zephyr-1/zephyr/drivers/wifi/nrfwifi/src/fmac_main.c:870:20: warning: initialization of 'int (*)(void *, const unsigned char *, enum wpa_alg,  const unsigned char *, int,  int,  const unsigned char *, size_t,  const unsigned char *, size_t,  enum key_flag)' {aka 'int (*)(void *, const unsigned char *, enum wpa_alg,  const unsigned char *, int,  int,  const unsigned char *, unsigned int,  const unsigned char *, unsigned int,  enum key_flag)'} from incompatible pointer type 'int (*)(void *, const unsigned char *, enum wpa_alg,  const unsigned char *, int,  int,  const unsigned char *, size_t,  const unsigned char *, size_t)' {aka 'int (*)(void *, const unsigned char *, enum wpa_alg,  const unsigned char *, int,  int,  const unsigned char *, unsigned int,  const unsigned char *, unsigned int)'} [-Wincompatible-pointer-types]
  870 |         .set_key = nrf_wifi_wpa_supp_set_key,
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/nordic/Documents/zephyr-1/zephyr/drivers/wifi/nrfwifi/src/fmac_main.c:870:20: note: (near initialization for 'wpa_supp_ops.set_key')
```

Reply:
Yes, already fixed it locally, was just waiting for any other comments before pushing.